app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/generate_when/TravelCertificateDateInputViewModel.kt

```diff
@@ -0,0 +1,270 @@
+
+    var travelDate by remember {
+      mutableStateOf(
+        Clock.System.now().toLocalDateTime(TimeZone.currentSystemDefault()).date,
+      )
+    }
+
+    var contractIdFromBackend by remember { mutableStateOf<String?>(null) }
+
+    var email by remember { mutableStateOf<String?>(null) }
+
+    var primaryInput by remember {
+      mutableStateOf<TravelCertificatePrimaryInput?>(null)
+    }
+
+    var isInputValid by remember { mutableStateOf<Boolean?>(null) }
+
+    var hasCoEnsured by remember { mutableStateOf(false) }
+
+    CollectEvents { event ->
+      when (event) {
+        TravelCertificateDateInputEvent.RetryLoadData -> {
+          loadIteration++
+        }
+
+        is TravelCertificateDateInputEvent.ChangeEmailInput -> {
+          email = event.email
+        }
+
+        is TravelCertificateDateInputEvent.ChangeDataInput -> {
+          travelDate = event.localDate
+        }
+
+        TravelCertificateDateInputEvent.NullifyInputValidity -> {
+          isInputValid = null
+        }
+
+        TravelCertificateDateInputEvent.ValidateInputAndChooseDirection -> {
+          val currentEmail = email
+          val isCurrentInputValid =
+            currentEmail != null && currentEmail.length > 5 // todo: couldn't be shorter than this, right?
+          isInputValid = isCurrentInputValid
+          if (isCurrentInputValid) {
+            if (hasCoEnsured) {
+              val currentContractId = contractIdFromBackend
+              val travelCertificatePrimaryInput = if (currentEmail != null &&
+                currentContractId != null
+              ) {
+                TravelCertificatePrimaryInput(
+                  currentEmail,
+                  travelDate,
+                  currentContractId,
+                )
+              } else {
+                null
+              }
+              if (travelCertificatePrimaryInput == null) {
+                logcat(LogPriority.INFO) {
+                  "TravelCertificateDateInputPresenter: currentEmail or currentContractId are null when they shouldn't be"
+                }
+              }
+              primaryInput = travelCertificatePrimaryInput
+            } else {
+              generateIteration++ //
+            }
+          } // todo: else isInputValid will be false will go to Success and show toast
+        }
+      }
+    }
+
+    LaunchedEffect(generateIteration) {
+      val currentContent = screenContent
+      val currentEmail = email
+      if (currentContent is DateInputScreenContent.Success && currentEmail != null) {
+        screenContent = DateInputScreenContent.Loading
+        createTravelCertificateUseCase.invoke(
+          contractId = currentContent.details.contractId,
+          startDate = travelDate,
+          isMemberIncluded = true,
+          coInsured = listOf(),
+          email = currentEmail,
+        ).fold(
+          ifLeft = { _ ->
+            screenContent = DateInputScreenContent.Failure
```

Comment:
I actually thought it would probably be a good thing not to have this kind of retry? Because it could be user's fault (for example, invalid email), and if we just try to generate the URL with the same input as before on retry, it doesn't really solve the issue. And without the succefully downloaded url, we cannot navigate to overview anyway. It could be a good thing to change the text in the Failure Screen info, for example, if it is an invalid email? Similar as it happens in TravelCertificateTravellersInput now.